src/main/java/codechicken/nei/api/ShortcutInputHandler.java

```diff
@@ -0,0 +1,119 @@
+package codechicken.nei.api;
+
+import codechicken.nei.ItemPanels;
+import codechicken.nei.LayoutManager;
+import codechicken.nei.NEIClientConfig;
+import codechicken.nei.NEIClientUtils;
+import codechicken.nei.PositionedStack;
+import codechicken.nei.recipe.GuiCraftingRecipe;
+import codechicken.nei.recipe.GuiRecipe;
+import codechicken.nei.recipe.GuiUsageRecipe;
+import java.util.List;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.item.ItemStack;
+import org.lwjgl.input.Mouse;
+
+public class ShortcutInputHandler {
```

Comment:
Since it is more like a Utilities class a good thing would be to make it abstract
```suggestion
public abstract class ShortcutInputHandler {
```